src/puppetlabs/puppetdb/status.clj

```diff
@@ -0,0 +1,39 @@
+(ns puppetlabs.puppetdb.status
+  (:require [puppetlabs.puppetdb.scf.storage-utils :as sutils]
+            [puppetlabs.puppetdb.utils :as utils]
+            [puppetlabs.puppetdb.mq :as mq]
+            [puppetlabs.puppetdb.schema :as pls]
+            [schema.core :as s]))
+
+(def status-details-schema {:maintenance_mode? s/Bool
+                            :queue_depth s/Int
+                            :read_db_up? s/Bool
+                            :write_db_up? s/Bool})
+
+(pls/defn-validated status-details :- status-details-schema
+  "Returns a map containing status information on the various parts of
+  a running PuppetDB system. This data can be interpreted to determine
+  whether the system is considered up"
+  [config :- {s/Any s/Any}
+   shared-globals-fn :- pls/Function
+   maint-mode-fn? :- pls/Function]
+  (let [globals (shared-globals-fn)]
+    {:maintenance_mode? (maint-mode-fn?)
+     :queue_depth (utils/nil-on-failure
+                   (->> [:command-processing :mq :endpoint]
```

Comment:
Pre-existing but while we're moving this could you put this with the get-in below `(get-in config [...])`?